setup-hooks/strip: add stripExclude (#218783)
NixOS/nixpkgs: doc/stdenv/stdenv.chapter.md

Flags passed to the `strip` command applied to the files in the directories listed in `stripDebugList`. Defaults to `-S` (i.e. `--strip-debug`).

##### `stripExcludeExtensions` {#var-stdenv-stripExcludeExtensions}

> A space-separated list of file extensions to avoid stripping. E.g. setting this to `rlib` will exclude all `*.rlib` files from being stripped.

Reviewer: ```suggestion
A space-separated list of file extensions to avoid stripping.

The following example excludes all `*.rlib` files from being stripped:

```nix
stdenv.mkDerivation {
  # ...
  stripExcludeExtension = [ "rlib" ]
  # ...
}
```

Not sure if the example is meaningful, but in any case please provide a complete example without placeholders. The fact that I'm not sure whether mine is right hints at that important information is missing.

Author: Done. Technically I think `[ "foo" "bar"]` will have the same effect as `[ "foo bar" ]` because of bash, but I'd prefer to document the former since that's the intent (and maybe with structuredattrs the latter will stop working).
